client/src/main/java/com/tigrisdata/db/client/search/FacetFieldOptions.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022 Tigris Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tigrisdata.db.client.search;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tigrisdata.db.client.JSONSerializable;
+import java.util.HashMap;
+import java.util.Map;
+
+enum FacetFieldType {
+  value("value");
+  private final String strVal;
+
+  FacetFieldType(String v) {
+    this.strVal = v;
+  }
+
+  @Override
+  public String toString() {
+    return this.strVal;
+  }
+}
+
+public class FacetFieldOptions implements JSONSerializable {
+  private final FacetFieldType type;
+  private final long limit;
+
+  private static final long DEFAULT_LIMIT = 10;
+  private static final FacetFieldType DEFAULT_TYPE = FacetFieldType.value;
+  private static final FacetFieldOptions DEFAULT_INSTANCE =
+      new FacetFieldOptions(DEFAULT_TYPE, DEFAULT_LIMIT);
+
+  private FacetFieldOptions(FacetFieldType type, long limit) {
+    this.type = type;
+    this.limit = limit;
+  }
+
+  public FacetFieldType getType() {
+    return type;
+  }
+
+  public long getLimit() {
+    return limit;
+  }
+
+  public static FacetFieldOptions getDefaultInstance() {
+    return DEFAULT_INSTANCE;
+  }
+
+  @Override
+  public String toJSON(ObjectMapper objectMapper) {
+    Map<String, String> map =
+        new HashMap<String, String>() {
+          {
+            put("type", type.toString());
+            put("limit", String.valueOf(limit));
+          }
+        };
+
+    try {
+      return objectMapper.writeValueAsString(map);
+    } catch (JsonProcessingException e) {
+      throw new IllegalArgumentException("Failed to serialize FacetFieldOption to JSON", e);
+    }
+  }
+
+  public Builder newBuilder() {
+    return new Builder();
+  }
+
+  static class Builder {
+    private FacetFieldType type;
+
+    private long limit;
+
+    private Builder() {
+      this.limit = DEFAULT_LIMIT;
+    }
+
+    public Builder type(FacetFieldType type) {
```

Comment:
nit: I have been following `withXXX` convention wherever applicable for Builders. it will be good to keep it consistent.